dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ResourcesService.java

```diff
@@ -87,7 +87,7 @@
      * @return create directory result
      */
     @Transactional(rollbackFor = RuntimeException.class)
-    public Result createDirectory(User loginUser,
+    public synchronized Result createDirectory(User loginUser,
```

Comment:
Hi,
We just need make the same dir name synchronized, put synchronized in function will make createDirectory of  this model synchronized, it will cause bad performance.
Is there better way to fix this bug?

Reply:
Agree wtth yangyichao-mango,we also need also ensure the concurrent

Reply:
Ok
I will test it later.